### PR TITLE
 CIRCSTORE-220: Upgrade to raml-module-builder (RMB) 30.2.4 

### PR DESCRIPTION
**Pull request by julianladisch**
Updating RMB from 30.0.1 to RMB 30.2.4 fixes
https://issues.folio.org/browse/CIRCSTORE-210 "Requests -Filtering
by Request Type returns (Total Records = 1000) when there are more"
because it includes these fixes:

* [RMB-637](https://issues.folio.org/browse/RMB-637) Run ANALYZE after index (re-)creation
* [RMB-645](https://issues.folio.org/browse/RMB-645) Use where-only clause for the "count query"
* [RMB-639](https://issues.folio.org/browse/RMB-639) Unexpected breaking changes streamGet and getWithOptimizedSql

https://github.com/folio-org/raml-module-builder/blob/master/doc/upgrading.md#version-302 advises:
> https://issues.folio.org/browse/RMB-652 error message "may not be null" changed to "must not be null" (hibernate-validator)

### CODE REVIEW

**Comment by julianladisch** [comment]
This pull request is blocked by merge of #250

File: src/main/java/org/folio/rest/impl/LoanPoliciesAPI.java [hunk 1]
```diff
@@ -110,7 +113,103 @@ public void putLoanPolicyStorageLoanPoliciesByLoanPolicyId(
...
+        )
+        ||
+        ( loanPolicy.getRenewalsPolicy() != null
+          &&
+          ( ! is(loanPolicy.getRenewable()) ||
+            ! is(loanPolicy.getRenewalsPolicy().getDifferentPeriod())
+          )
+          && loanPolicy.getRenewalsPolicy().getAlternateFixedDueDateScheduleId() != null
+       )) {
+      String message =
```
Comment (craigmcnally) [inline src/main/java/org/folio/rest/impl/LoanPoliciesAPI.java:154]: Consider using a stringBuilder here

File: src/main/java/org/folio/rest/impl/LoanPoliciesAPI.java [hunk 2]
```diff
@@ -110,7 +113,103 @@ public void putLoanPolicyStorageLoanPoliciesByLoanPolicyId(
...
+        loanPolicy.getLoanable() + " and profile is of type fixed";
+      respond422("fixedDueDateScheduleId",
+          loanPolicy.getLoansPolicy().getFixedDueDateScheduleId(),
+          message, asyncResultHandler);
+      return;
+    }
+
+    // alternate renewal loan period
+    if (isFixed
+        && is(loanPolicy.getRenewable())
```
Comment (craigmcnally) [inline src/main/java/org/folio/rest/impl/LoanPoliciesAPI.java:188]: The formatting of these large conditionals is inconsistent, please pick a format and use it throughout

Comment (julianladisch) [reply src/main/java/org/folio/rest/impl/LoanPoliciesAPI.java:188]: The formatting is exactly the formatting of the *.drl files that are to be deleted.

Comment (julianladisch) [reply src/main/java/org/folio/rest/impl/LoanPoliciesAPI.java:188]: The formatting is kept to make code review easy.

Comment (marcjohnson-kint) [reply src/main/java/org/folio/rest/impl/LoanPoliciesAPI.java:188]: Personally, from my reviewing perspective, I'm not comparing this code to the deleted drools files, I'm considering this as new code. I don't think the formatting needs to be retained.

**Comment by julianladisch** [comment]
Unblocking this pull request because master with #250 has been merged.

**Pull request merged**